src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategyCrutch.java

```diff
@@ -0,0 +1,60 @@
+package org.zalando.nakadi.partitioning;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class HashPartitionStrategyCrutch {
+
+    private static final String PROPERTY_PREFIX = "nakadi.hashPartitioning.overrideOrder";
+
+    // due to initial problem with order of partitions we need to keep the original order now
+    private final Map<Integer, List<Integer>> partitionsOrder;
+
+    @Autowired
+    @SuppressWarnings("unchecked")
+    public HashPartitionStrategyCrutch(final Environment environment,
+                                       @Value("${" + PROPERTY_PREFIX + ".max}") final int maxPartitionNum) {
+
+        final ImmutableMap.Builder<Integer, List<Integer>> mapBuilder = ImmutableMap.builder();
+        for (int pCount = 1; pCount <= maxPartitionNum; pCount++) {
+
+            final String propertyName = PROPERTY_PREFIX + ".p" + pCount;
+            final List<String> predefinedOrder = (List<String>) environment.getProperty(propertyName, List.class);
```

Comment:
I see that you do map to integer later. why don't you want to cast it to (List<Integer>) right here?